webapp/tests/forms/steps/test_personal_data_steps.py

```diff
@@ -192,6 +192,16 @@ def test_if_married_and_lived_separated_and_no_separated_since_given_then_fail_v
         self.assertFalse(form.validate())
         self.assertIn('familienstand_married_lived_separated_since', form.errors)
 
+    def test_if_married_and_lived_separated_and_separated_since_is_invalid_date_then_fail_validation(self):
+        data = MultiDict({'familienstand': 'married',
+                          'familienstand_date': '03.04.2008',
+                          'familienstand_married_lived_separated': 'yes',
+                          'familienstand_married_lived_separated_since': '99.99.9999',
+                          'familienstand_confirm_zusammenveranlagung': 'y'})
+        form = self.step.Form(formdata=data)
+        self.assertFalse(form.validate())
+        self.assertIn('familienstand_married_lived_separated_since', form.errors)
+
```

Comment:
Maybe we could add the same tests for the familienstand_date itself?

Reply:
Yes, good point, will do.